client/web/web.go

```diff
@@ -309,80 +300,52 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if r.URL.Path == "/redirect" || r.URL.Path == "/redirect/" {
+	switch {
+	case r.URL.Path == "/redirect" || r.URL.Path == "/redirect/":
 		io.WriteString(w, authenticationRedirectHTML)
 		return
+	case r.Method == "POST":
+		s.servePostNodeUpdate(w, r)
+		return
+	default:
+		s.serveGetNodeData(w, r, user, false)
+		return
 	}
+}
+
+type nodeData struct {
+	Profile           tailcfg.UserProfile
+	SynologyUser      string
+	Status            string
+	DeviceName        string
+	IP                string
+	AdvertiseExitNode bool
+	AdvertiseRoutes   string
+	LicensesURL       string
+	TUNMode           bool
+	IsSynology        bool
+	DSMVersion        int // 6 or 7, if IsSynology=true
+	IsUnraid          bool
+	UnraidToken       string
+	IPNVersion        string
+}
 
-	st, err := s.lc.Status(ctx)
+func (s *Server) serveGetNodeData(w http.ResponseWriter, r *http.Request, user string, asJSON bool) {
```

Comment:
I think overloading this method to handle both the old static index page, as well as the JSON raw data output for the new page is somewhat confusing.  

Maybe there is a single `getNodeData()` method that returns a `nodeData` struct.  And then separate handler methods for the old go-template index page and the new `/api/data` JSON endpoint, each of which call `getNodeData()` ?

Reply:
Figured we'd be getting rid of it once we cut out the old html templates, but I like your suggestion - a little cleaner in the interim.